hedera-mirror-rest/viewmodel/customFeeViewModel.js

```diff
@@ -26,60 +26,60 @@ class CustomFeeViewModel {
-      this.maximum = customFee.maximumAmount || undefined;
-      this.minimum = customFee.minimumAmount;
-      this.net_of_transfers = customFee.netOfTransfers;
-    } else if (customFee.royaltyDenominator) {
-      // royalty fee
-      this.amount = {
-        numerator: customFee.royaltyNumerator,
-        denominator: customFee.royaltyDenominator,
-      };
-
-      if (customFee.amount) {
-        this.fallback_fee = this._parseFixedFee(customFee);
-      }
-    } else {
-      // fixed fee
-      Object.assign(this, this._parseFixedFee(customFee));
-    }
-
-    this.collector_account_id = EntityId.parse(customFee.collectorAccountId, {isNullable: true}).toString();
+    this.fixed_fees = customFee.fixedFees?.map((f) => this._parseFixedFee(f)) ?? [];
+    this.fractional_fees = customFee.fractionalFees?.map((f) => this._parseFractionalFee(f, customFee.tokenId)) ?? [];
+    this.royalty_fees = customFee.royaltyFees?.map((f) => this._parseRoyaltyFee(f)) ?? [];
   }
 
-  hasFee() {
-    return !!this.amount;
-  }
-
-  isFixedFee() {
-    return this.amount.denominator === undefined;
+  _parseFixedFee(fixedFee) {
+    return {
+      all_collectors_are_exempt: fixedFee.allCollectorsAreExempt ?? false,
+      amount: fixedFee.amount,
+      collector_account_id: EntityId.parse(fixedFee.collectorAccountId, {isNullable: true}).toString(),
+      denominating_token_id: EntityId.parse(fixedFee.denominatingTokenId, {isNullable: true}).toString(),
+    };
   }
 
-  isFractionalFee() {
-    return this.amount.denominator !== undefined && this.minimum !== undefined;
+  _parseFractionalFee(fractionalFee, tokenId) {
+    return {
+      all_collectors_are_exempt: fractionalFee.allCollectorsAreExempt ?? false,
+      amount: {
+        numerator: fractionalFee.amount,
+        denominator: fractionalFee.amountDenominator,
+      },
+      collector_account_id: EntityId.parse(fractionalFee.collectorAccountId, {isNullable: true}).toString(),
+      denominating_token_id: EntityId.parse(tokenId, {isNullable: true}).toString(),
+      maximum: fractionalFee.maximumAmount,
+      minimum: fractionalFee.minimumAmount,
+      net_of_transfers: fractionalFee.netOfTransfers ?? false,
+    };
   }
 
-  isRoyaltyFee() {
-    return this.amount.denominator !== undefined && this.minimum === undefined;
-  }
+  _parseRoyaltyFee(royaltyFee) {
+    const fallbackFee = royaltyFee.fallbackFee?.amount
+      ? {
+          amount: royaltyFee.fallbackFee.amount,
+          denominating_token_id: EntityId.parse(royaltyFee.fallbackFee.denominatingTokenId, {
+            isNullable: true,
+          }).toString(),
+        }
+      : null;
 
-  _parseFixedFee(customFee) {
-    return {
-      amount: customFee.amount,
-      denominating_token_id: EntityId.parse(customFee.denominatingTokenId, {isNullable: true}).toString(),
+    const viewModel = {
+      all_collectors_are_exempt: royaltyFee.fallbackFee?.allCollectorsAreExempt ?? false,
```

Comment:
```suggestion
    return {
      fallback_fee,
      all_collectors_are_exempt: royaltyFee.fallbackFee?.allCollectorsAreExempt ?? false,
```

note please add `fallback_fee` after `collector_account_id`
 
if `fallback_fee` is null we show `"fallback_fee": null` in the json response. so the function shouldn't conditionally show `fallback_fee`.

Reply:
Thanks I have added the null for missing fallback fees.